### PR TITLE
[DOCS] k0smotron edits for FAQ and Troubleshooting topic creation.

### PR DESCRIPTION
**Pull request by KoryKessel-Mirantis**
k0smotron edits for FAQ and Troubleshooting topic creation.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ nav:
         - Docker: capi-docker.md
         - vSphere: capi-vsphere.md
   - FAQ: faq.md
+  - FAQ: troubleshooting.md
```
Comment (MagdaDziadosz) [inline mkdocs.yml:39]:
```suggestion
  - Troubleshooting: troubleshooting.md
```

**Review by MagdaDziadosz — CHANGES_REQUESTED** [review]

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+
+### I cannot join a worker node
+
+In [standalone](usage-overview.md#standalone) mode, check whether the join
+token has expired. If this is the case, attempt to create a new one using
+[JoinTokenRequest](https://docs.k0smotron.io/stable/join-nodes/#join-tokens) In
```
Comment (makhov) [inline docs/troubleshooting.md:6]: Can we make the sentences “In standalone...” and “In Cluster API...” in separate paragraphs?

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+[Cluster API](cluster-api.md), check the logs of your infrastructure provider controller.
+
+Also, check to see whether different versions were used in the initial cluster
+creation for the control plane and the worker nodes, as Kubernetes allows a
+limited [version skew](https://kubernetes.io/releases/version-skew-policy/).
+
+### MachineDeployment does not function
```
Comment (makhov) [inline docs/troubleshooting.md:13]: "MachineDeployment with Docker Provider does not function" or smth. This item is specific to Docker Provider

File: docs/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+
+### I cannot join a worker node
+
+In [standalone](usage-overview.md#standalone) mode, check whether the join
+token has expired. If this is the case, attempt to create a new one using
+[JoinTokenRequest](https://docs.k0smotron.io/stable/join-nodes/#join-tokens) In
+[Cluster API](cluster-api.md), check the logs of your infrastructure provider controller.
+
+Also, check to see whether different versions were used in the initial cluster
```
Comment (makhov) [inline docs/troubleshooting.md:9]: Maybe use "compatible" here? Kubernetes is ok with a limited number of different versions.

File: docs/troubleshooting.md (hunk 2 above)
Comment (KoryKessel-Mirantis) [reply docs/troubleshooting.md:6]: Yes.

File: docs/troubleshooting.md (hunk 3 above)
Comment (KoryKessel-Mirantis) [reply docs/troubleshooting.md:13]: Added 'with Docker Provider'.

File: docs/troubleshooting.md (hunk 4 above)
Comment (KoryKessel-Mirantis) [reply docs/troubleshooting.md:11]: Done.

**Review by makhov — APPROVED** [review]

File: docs/faq.md [hunk 5]
```diff
@@ -1,15 +1,62 @@
...
+    controller manager, all of the related CRD definitions, and the necessary
+    RBAC rules. This approach ensures a consistent configuration across
+    clusters and simplifies maintenance and management tasks.
+
+### What is the relation of k0smotron to Cluster API?
+
+k0smotron is a fully compliant [Cluster API](https://cluster-api.sigs.k8s.io/)
+provider for [k0s](https://k0sproject.io/) that can be used with any Cluster
+API compatible tooling. In addition, k0smotron is a Cluster API infrastructure
+provider which you can usse SSH connections to provision clusters on remote
```
Comment (MagdaDziadosz) [inline docs/faq.md:43]:
```suggestion
provider which you can use SSH connections to provision clusters on remote
```

File: docs/faq.md [hunk 6]
```diff
@@ -1,15 +1,62 @@
...
+machines.
+
+### What do we meant by "from pets to cattle"?
+
+A _pets_ service model describes carefully tended resources that are nurtured
```
Comment (MagdaDziadosz) [inline docs/faq.md:48]:
```suggestion
A *pets* service model describes carefully tended resources that are nurtured
```

**Review by MagdaDziadosz — CHANGES_REQUESTED** [review]

**Review by MagdaDziadosz — APPROVED** [review]

**Pull request merged**